tools/ingest_historical_csv.py

```diff
@@ -105,15 +104,15 @@ async def _main(args):
     func, url = _get_executor(args)
     sem = asyncio.BoundedSemaphore(args.concurrency)
     async with ClientSession() as session:
-        tasks = []
         for csv in args.csv:
             log.debug('Convert records in {} to JSON'.format(csv))
             df = pd.read_csv(csv, index_col=None, header=0)  # pylint: disable=invalid-name
             log.debug('CSV Record count {count}'.format(count=len(df)))
-            updated_df = _update_df(df, csv)
+            updated_df = _update_df(df)
+            # Runnning one file at a time to overcome duplicate issue for similar record across different ecosystem
             task = asyncio.ensure_future(func(df=updated_df, session=session, url=url, csv=csv, sem=sem))
-            tasks.append(task)
-        _ = await asyncio.gather(*tasks)
+            _ = await asyncio.gather(task)
+
```

Comment:
This logic defeats the purpose of async io. This change would wait for the completion before making next request.

Reply:
Updated logic as we shouldn't insert duplicate data if similar data exist in different ecosystem due to gremlin limitation. We are executing single file data at a time, but all data will insert parallel through asyncio.